### DISCUSSION TITLE
Feature-Request: Clear window when video is downloaded and revers ordering for multiple videos per tab

### DISCUSSION THREAD

**Discussion by ttrepper** [aclap-dev/video-downloadhelper#159] [Help and Q&A]
### Your question

1) Clear video-windows when a video is downloaded
I would like to request a new checkbox in the apprearance settings that - when ticked - will clear the list of videos in the video window. Reason: When a user is surfing within a single browser tab and downloads a video and surfes to another page wihin the same browser tab and downloads another video this list will become dirty. This setting should clear the list of videos found when a video is downloaded. 

2) Revers-ordering of videolist with newest on top instead of bottom
I would like to request another checkbox in the apprearance settings please that - when ticked - should order the videos in the list revers with the newest on top instead of bottom which is the actual behaviour. Reason: When downloading multiple videos from a page showing multiple videos all happens within the same browser tab and it would be more convenient to have the choice of having the lasted ones found on top or at the bottom. 

Thank you very much in advance. 

### Addon & CoApp version and details

version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: de
license: accepted
platform: aarch64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "arm64",
    "node": "18"
  },
  "home": "/Users/myPath",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Comment by GrampaWildWilly** [comment]
Interesting ideas.  I can see their usefulness.  I have often been a bit annoyed that when I reload a page, VDH shows me 2 sets of the same variants.  And another reload would give me 3.  And a new set with each reload.  I would prefer if a page reload wiped the menu & then put the new contents into a previously empty list, same as it does on your first visit to the page.  After you reload a page, I suspect the previous list of variants is no longer valid anyway.

Now, there's a few things that might be tricky about the sorting.  VDH gives you the variants in descending order of resolution.  If you have multiple videos on the page, you're going to get blocks of variants, each block showing a list of variants in descending order of resolution.  I would imagine that you would want your sorting ideas to apply to the blocks of variants, while you would still want the variants within a given block to be presented in descending order by resolution.

Just to be clear, I don't work for VDH.  I'm just another user, like you.
